File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/IRequestDataObject.cs


#region "Imports"

#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Interface for all request data objects.
	/// </summary>
	/// <remarks>This interface can be used to create a new 
	/// request data object.</remarks>
	public interface IRequestDataObject
	{
		#region "Functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		/// <remarks>When implemented in the derived class, this method 
		/// should be preferred to be made as an internal method.</remarks>
		void GenerateRequest();

		#endregion
	}

}

File: java/src/sdk/dataobjects/paypal/payflow/IRequestDataObject.java
package paypal.payflow;

/**
 * Interface for all request data objects.
 * <p/>
 * This interface can be used to create a new request data object.
 * </p>
 */

public interface IRequestDataObject {

    /**
     * /// Generates the transaction request.
     * /// <p>When implemented in the derived class, this method
     * /// should be preferred to be made as an internal method.</p>
     */
    void generateRequest();

}

